src/prefect/client/orion.py

```diff
@@ -1305,8 +1305,8 @@ async def create_deployment(
             parameters=dict(parameters or {}),
             tags=list(tags or []),
             work_queue_name=work_queue_name,
-            worker_pool_name=worker_pool_name,
-            worker_pool_queue_name=worker_pool_queue_name,
+            work_pool_name=work_pool_name,
+            work_pool_queue_name=work_pool_queue_name,
```

Comment:
I think this change will result in version incompatibilities like we had between 2.7.6 and 2.7.7. We'll get 422 errors when trying to create deployments for servers and clients with different versions due to the extra fields. I'm not sure if this can be avoided, but we might want to find a way to maintain version compatibility. Maybe we could use a validator on `DeploymentCreate`?